tls/s2n_client_key_exchange.c

```diff
@@ -127,11 +136,31 @@ int s2n_rsa_client_key_recv(struct s2n_connection *conn, struct s2n_blob *shared
     conn->secure.rsa_premaster_secret[0] = client_hello_protocol_version[0];
     conn->secure.rsa_premaster_secret[1] = client_hello_protocol_version[1];
 
+    S2N_ASYNC_PKEY_DECRYPT(conn, &encrypted, shared_key, s2n_rsa_client_key_recv_complete);
+}
+
+int s2n_rsa_client_key_recv_complete(struct s2n_connection *conn, int rsa_failed, struct s2n_blob *decrypted)
+{
+    S2N_ERROR_IF(decrypted->size != S2N_TLS_SECRET_LEN, S2N_ERR_SIZE_MISMATCH);
```

Comment:
Do you want to migrate all your new `S2N_ERROR_IF` to `ENSURE_POSIX`?

Reply:
Was thinking to keep the previous code as it is... I can do it for the things that I touch, but the code will kind of be mixed then.